File: atividade-4/038_rotDir/038_rotDir.hs
module Main where

--IN : Um natural n e uma lista ou string S
--OUT: Lista S rotacionada n vezes à direita

rotDir :: Int -> String -> String
rotDir 0 xs = xs
rotDir n xs = rotDir (n - 1) (last xs : init xs)

main = do
  a <- readLn :: IO Int
  b <- getLine
  print $ rotDir a b

File: atividade-4/038_rotDir/038_rotDir.js
//IN : Um natural n e uma lista ou string S
//OUT: Lista S rotacionada n vezes à direita

const n = 4;
const w = "asdfg";

const init = (xs) => xs.slice(0, xs.length - 1);
const last = (xs) => xs[xs.length - 1];

const rotDir = (n, w) => {
  if (!n) return w;

  return rotDir(n - 1, last(w).concat(init(w)));
};

console.log(rotDir(n, w));
